editor/plugins/tile_map_editor_plugin.cpp

```diff
@@ -220,22 +220,23 @@ void TileMapEditor::set_selected_tiles(Vector<int> p_tiles) {
 	palette->ensure_current_is_visible();
 }
 
-void TileMapEditor::_create_set_cell_undo(const Vector2 &p_vec, const CellOp &p_cell_old, const CellOp &p_cell_new) {
+Dictionary TileMapEditor::_create_cell_dictionary(int tile, bool flip_x, bool flip_y, bool transpose, Vector2 autotile_coord) {
+
+	Dictionary cell;
 
-	Dictionary cell_old;
-	Dictionary cell_new;
+	cell["id"] = tile;
+	cell["flip_h"] = flip_x;
+	cell["flip_y"] = flip_y;
+	cell["transpose"] = transpose;
+	cell["auto_coord"] = autotile_coord;
 
-	cell_old["id"] = p_cell_old.idx;
-	cell_old["flip_h"] = p_cell_old.xf;
-	cell_old["flip_y"] = p_cell_old.yf;
-	cell_old["transpose"] = p_cell_old.tr;
-	cell_old["auto_coord"] = p_cell_old.ac;
+	return cell;
+}
+
+void TileMapEditor::_create_set_cell_undo(const Vector2 &p_vec, const CellOp &p_cell_old, const CellOp &p_cell_new) {
```

Comment:
maybe this functions should be called _create_set_cell_undo_redo

Reply:
Done, function renamed.